apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -82,30 +80,10 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 		return domainObjects;
 	}
 
-	public async findAvailableByUser(user: UserDO, skip = 0, limit = 1000000, nameQuery?: string): Promise<Group[]> {
-		let nameQueryMatch = {};
-		if (nameQuery && StringValidator.isNotEmptyString(nameQuery, true)) {
-			const escapedName = nameQuery.replace(MongoPatterns.REGEX_MONGO_LANGUAGE_PATTERN_WHITELIST, '').trim();
-			if (StringValidator.isNotEmptyString(escapedName, true)) {
-				nameQueryMatch = { name: { $regex: escapedName, $options: 'i' } };
-			}
-		}
-
-		const pipeline = [
-			{ $match: { users: { $elemMatch: { user: new ObjectId(user.id) } } } },
-			{ $match: nameQueryMatch },
-			{
-				$lookup: {
-					from: 'courses',
-					localField: '_id',
-					foreignField: 'syncedWithGroup',
-					as: 'syncedCourses',
-				},
-			},
-			{ $match: { syncedCourses: { $size: 0 } } },
-			{ $skip: skip },
-			{ $limit: limit },
-		];
+	public async findAvailableByUser(user: UserDO, skip?: number, limit?: number, nameQuery?: string): Promise<Group[]> {
```

Comment:
Why you have removed the default values and mate it optional instead? Where is findAvailableByUser used? Could it break something?

Reply:
the find available functions were introduced in this PR, so no one else is using this atm. I removed the defaults, to make checks in a private method as suggested by @MarvinOehlerkingCap